pkg/http/sse.go

```diff
@@ -0,0 +1,97 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/go-chi/render"
+)
+
+type StreamAdapter interface {
+	// ResponseProvider returns the response to be sent back to client.
+	// Any errors that occur should be handled and written to `w`, returning false as `ok`.
+	ResponseProvider(w http.ResponseWriter, r *http.Request) (response interface{}, ok bool)
```

Comment:
1) `ResponseProvider` doesn't sound like a method name, I would rather expect `ProvideResponse`?
2) Wouldn't something like `ResponseProvider(r *http.Request) (response interface{}, err error)` make more sense? On the other hand, it would mean that Watermill is responsible for marshalling the error response, which may be a bit too opinionated

Reply:
1. Right, that sounds more like an interface. I'm still looking for better names for those.
2. Maybe that's related to your previous comment - if we introduced another method for marshalling errors (and provide a default implementation), we could get rid of the `writer` in this one.

Reply:
it could be maybe a *little bit* prettier (but more complex), if `response` wasn't an `interface{}`, but rather something along the lines of `interface{ Respond(w http.ResponseWriter) }`.
It *is* kind of opinionated though, that we impose on the implementer of `StreamAdapter` the exact method that whatever `ResponseProvider` returns needs to implement.

Reply:
The channel from chi/render accepts `interface{}`, so that probably would not change much.